packages/amplify-e2e-core/src/utils/sdk-calls.ts

```diff
@@ -204,6 +204,38 @@ export const describeCloudFormationStack = async (stackName: string, region: str
   );
 };
 
+export const getNestedStackID = async (stackName: string, region: string, logicalId: string) => {
+  const cfnClient = new CloudFormation({ region });
+  const resource = await cfnClient.describeStackResources({ StackName: stackName, LogicalResourceId: logicalId }).promise();
+  return resource?.StackResources?.[0] ?? null;
+};
+
+/**
+ * Collects table resource id from parent stack
+ * @param region region the stack exists in
+ * @param table name of the table used in the appsync schema
+ * @param StackId id of the parent stack
+ * @returns
+ */
+
+export const getTableResourceId = async (region: string, table: string, StackId: string): Promise<string | null> => {
+  const cfnClient = new CloudFormation({ region });
+  const apiResources = await cfnClient
+    .describeStackResources({
+      StackName: StackId,
+    })
+    .promise();
+  const resource = apiResources.StackResources.find((stackResource) => table === stackResource.LogicalResourceId);
+  const tableStack = await cfnClient.describeStacks({ StackName: resource.PhysicalResourceId }).promise();
```

Comment:
We should check if `resource` is `undefined` or not in case `find` did not return anything.